code-submissions-clustering-core/src/main/kotlin/org/jetbrains/research/code/submissions/clustering/util/psi/PsiUtil.kt

```diff
@@ -0,0 +1,26 @@
+package org.jetbrains.research.code.submissions.clustering.util.psi
+
+import com.intellij.openapi.command.WriteCommandAction
+import com.intellij.psi.PsiElement
+import com.intellij.psi.codeStyle.CodeStyleManager
+import org.jetbrains.research.code.submissions.clustering.load.context.builder.gumtree.converter.PsiToGumTreeConverter.filterWhiteSpaces
+
+val PsiElement.isLeaf: Boolean
+    get() = this.children.isEmpty()
```

Comment:
Should we use getElementChildren here?

Reply:
I guess, this implementation is just fine, getElementChildren will cause unnecessary memory allocation and complicate the logic.